src/main/java/hw6/DataFromJsonForMetalsColorPage.java

```diff
@@ -0,0 +1,79 @@
+package hw6;
+
+import hw6.entities.Summary;
+
+import java.util.List;
+import java.util.Objects;
+
+public class DataFromJsonForMetalsColorPage {
```

Comment:
This class has to be renamed.
You should not mix two different layers even in class name. Class has no idea where data was stored and fwhat kind of file format used to. 
Check Entity Driving Approach in class work please. Example with User and Login Form.